### PR TITLE
LUCENE-9229 fixed 3 javadoc links 1) google guave, 2&3) Two links relocated away from "aggregateknowledge"

### PR DESCRIPTION
**Pull request by kmheibel**
https://issues.apache.org/jira/browse/SOLR-XXXXX

# Description

Please provide a short description of the changes you're making with this pull request.

# Solution

Please provide a short description of the approach taken to implement your solution.

# Tests

Please describe the tests you've developed or run to confirm this patch implements the feature or solves the problem.

# Checklist

Please review the following and check all that apply:

- [ ] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [ ] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Comment by epugh** [comment]
Thanks for this...   I know this is a small PR, but if you want to get in the habit of having the PR Description be filled in, that always helps folks!   The link back to the JIRA issue and just update the Description/Solution/Test sections.   a "not applicable" is fine!

**Comment by kmheibel** [comment]
https://issues.apache.org/jira/browse/LUCENE-9229

Fixed 3 broken links in the javadoc

1) A link to guava relocated by google.

2) Two links relocated away from "aggregateknowledge".

File: solr/core/src/java/org/apache/solr/util/hll/HLL.java [hunk 1]
```diff
@@ -44,10 +44,10 @@
  *
  * This data structure is interoperable with the implementations found at:
  * <ul>
- *   <li><a href="https://github.com/aggregateknowledge/postgresql-hll">postgresql-hll</a>, and</li>
+ *   <li><a href="https://github.com/citusdata/postgresql-hll/blob/master/README.md">postgresql-hll</a>, and</li>
  *   <li><a href="https://github.com/aggregateknowledge/js-hll">js-hll</a></li>
  * </ul>
- * when <a href="https://github.com/aggregateknowledge/postgresql-hll/blob/master/STORAGE.markdown">properly serialized</a>.
+ * when <a href="https://github.com/metdos/postgresql-hll/blob/master/STORAGE.markdown">properly serialized</a>.
```
Comment (janhoy) [inline solr/core/src/java/org/apache/solr/util/hll/HLL.java:50]: It feels a bit strange that we have a July-2015 fork of the hll library in our code base, and then link to live, changing documentation for two different forks of the original library. 

Better link to the citrusdata repo and a specific version back in time, like https://github.com/citusdata/postgresql-hll/tree/v2.9.0 which is the latest version that seems to have STORAGE.markdown. May not be 100% exact but close enough.

**Comment by janhoy** [comment]
Thanks. For future PRs, please fill in the description field and complete the checklist. The title of the PR should start with the JIRA issue number, e.g. "SOLR-1234 my feature".

For minor dead-link fixes like these we don't need individual JIRA issues, but for future larger works, separate JIRAs can be created.

**Pull request merged**